guide/english/computer-science/garbage-collection/index.md

```diff
@@ -5,16 +5,16 @@ title: Garbage Collection
 
 #### What is Garbage Collection?
 
-In general layman term, Garbage collection (GC) is nothing but collecting or gaining the memory back which has been allocated to the objects, which is not in use at the moment in any part of our program. A brief description as below.
+In layman's terms, Garbage collection (GC) is nothing more than collecting or gaining the memory back which has been allocated to objects, which is not in use at the moment in any part of a program. A brief description is below.
 
-Garbage collection is the process in which programs try to free up memory space that is no longer used by objects, and such.  Garbage collection is implemented differently for every language.  Most high-level programming languages have some sort of garbage collection built it.  Low-level programming languages may add garbage collection through libraries.
+Garbage collection is the process in which programs try to free up memory space that is no longer used by objects.  Garbage collection is implemented differently for every language.  Most high-level programming languages have some sort of garbage collection built it.  Low-level programming languages may add garbage collection through libraries.
 
-As said above, every programming language has their own way of GC. In C programming, developers need to take care of memory allocation and deallocation using malloc() and dealloc() functions. But, in case of C# developers no need to take care of GC and it's not recommended as well.    
+As said above, every programming language has their own way of implementing GC. In C programming, developers need to take care of memory allocation and deallocation using malloc() and dealloc() functions. But, in case of C# developers do not need to take address GC and it's also not a recommended practice.    
  
  #### How memory allocation happening?
- In C#, memory allocation of objects are happened in managed heap. which takes care by CLR (common language runtime). Memory allocation for the heap is done through win32 dll in OS as like in the C. But, In C objects are placed in memory where ever the free space suits the size of object. And, the memory mapping is woks based on Linkedlist concepts. In C#, memory allocation for heap is happening in linear manner. like one after another. 
+ In C#, memory allocation of objects are addressed in managed heap. which takes care by CLR (common language runtime). Memory allocation for the heap is done through win32 dll in the Operating System just like in C. But, In C objects are placed in memory where ever the free space suits the size of object. And, the memory mapping is works based on Linkedlist concepts. In C#, memory allocation for heap is happening in a consecutive manner. 
```

Comment:
removing the word `works` will be better here: `memory mapping is works based` -> `memory mapping is based`
small grammatical change: `Is happening in a consecutive manner` -> `for heap occurs in a consecutive manner`.